src/LiveQueryClient.js

```diff
@@ -178,9 +178,9 @@ class LiveQueryClient extends EventEmitter {
    *
    * @param {Object} query - the ParseQuery you want to subscribe to
    * @param {string} sessionToken (optional)
-   * @return {Object} subscription
+   * @return {Promise<LiveQuerySubscription>} subscription
    */
-  subscribe(query: Object, sessionToken: ?string): Object {
```

Comment:
Since it is a breaking change, I was wondering if perhaps would be better to continue returning the subscription object. The ones interested in waiting it to resolve can just do subscription.subscribePromise.then(() => { ... });

Reply:
This isn't a breaking change as `query.subscribe()` already returns `Promise<LiveQuerySubscription>`. As you can see by the [integration tests](https://github.com/parse-community/Parse-SDK-JS/blob/master/integration/test/ParseLiveQueryTest.js) I didn't have to change anything.

Reply:
I think the `await` in the integration tests are unnecessary. It returns a LiveQuerySubscription object as you can see [here](https://github.com/parse-community/Parse-SDK-JS/blob/master/src/LiveQueryClient.js#L205)

Reply:
The docs do not show the `await` as well: http://docs.parseplatform.org/js/guide/#create-a-subscription

Reply:
Docs should be updated.

You have to look [query.subscribe()](https://github.com/parse-community/Parse-SDK-JS/blob/master/src/ParseQuery.js#L1611)

Reply:
so agree with you that `query.subscribe()` is currently already returning a `Promise` and the related docs should be fixed, by `client.subscribe()` is currently not returning a `Promise` and the [docs](http://docs.parseplatform.org/js/guide/#subscribe) are right. It will be a breaking change for the `client.subscribe()`. Maybe you should do the trick [here](https://github.com/parse-community/Parse-SDK-JS/blob/master/src/ParseQuery.js#L1620), leaving the client as is now and awaiting the subscription promise to resolve, before returning the subscription.

Reply:
Oh I see